src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1559,25 +1557,25 @@ void SolverInterfaceImpl::computeMappings(const utils::ptr_vector<MappingContext
   }
 }
 
-void SolverInterfaceImpl::mapData(DataContext *context, const std::string &mappingType)
+void SolverInterfaceImpl::mapData(DataContext &context, const std::string &mappingType)
 {
   PRECICE_TRACE();
   using namespace mapping;
   MappingConfiguration::Timing timing;
-  if (context->hasMapping()) {
-    timing         = context->mappingContext().timing;
-    bool hasMapped = context->mappingContext().hasMappedData;
+  if (context.hasMapping()) {
```

Comment:
Let's use a guard here to reduce indentation.

```cpp
  if (!context.hasMapping()) {
    return;
  }
```

Reply:
Done. 4664788f5619f89dac0ebaec49d1f6aa86f291e0